WordPress/gutenberg
# File block: Add color block support to File block

## What?

Add color block support to file block.
Sets text color to false, since the block does not use plain text.

Partial for [18989](https://github.com/WordPress/gutenberg/issues/18989)
Replaces [31465](https://github.com/WordPress/gutenberg/pull/31465)

## Why?

To allow customizing the block colors.

## Testing Instructions





1. In the block editor, add two file blocks, one with a PDF, one with a different file format.
2. Change the block colors.
3. Save and view the front. The colors in the editor and front should match.


## Screenshots or screencast 

After:
![link color and background color applied to the block](https://user-images.githubusercontent.com/7422055/175024161-7661f56d-8807-48e2-b028-865b0581f9e7.png) 


Note, without the `display: inline-block` on the button link, the background color does not cover the block:
![Background color only shows for part of the block](https://user-images.githubusercontent.com/7422055/175021855-117b960e-baeb-4c09-9ce1-c4c33ef40d0d.png)
